name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java

```diff
@@ -712,6 +714,7 @@ private void updateChart()
 
             if (security == null || security.getPrices().isEmpty())
             {
+                chart.resetAxes();
```

Comment:
Last thing (actually, I just see a second thing …): I would call `resetAxes()` once, before any `if` blocks, so that we always start from the same, known state. So above, at [lines 702–711](https://github.com/buchen/portfolio/blob/master/name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java#L702-L711), where everything else is cleared (marker lines, paint listeners etc.).